### ISSUE TITLE
Standard users are unable to provision RKE1/RKE2 clusters with PSACT due to being unable to view global PSACTs

### ISSUE THREAD

**Issue by markusewalker** [rancher/rancher#40288]
**Setup**
- Rancher version: `v2.7-head` https://github.com/rancher/rancher/commit/6316baf1697cca085e073ccd5e9e7a3bc542d0e5
- Browser type & version: Chrome

**Describe the bug**

When working with Pod Security Admission Configuration Templates (PSACT), it appears that only the admin user is able to do so successfully. In Cluster Management, below is the admin left view pane:

![image](https://user-images.githubusercontent.com/52591819/213767579-83b4e04d-4c15-48bd-9de1-99d2a1a9a586.png)

And here is the standard user view pane:

![image](https://user-images.githubusercontent.com/52591819/213767833-2c46a70d-8ac4-4267-acd1-25bb86a291ed.png)

Standard users need to be able to work with PSACT when provisioning clusters; adding a standard user as a Cluster Owner to an admin-provisioned cluster isn't a viable workaround. As a consequence of standard users not being able to view PSACT in Cluster Management, when provisioning a cluster, the following issue is seen:

![image](https://user-images.githubusercontent.com/52591819/213769575-1ba858cd-ed3f-45ad-920b-0e805628d51a.png)

I should be able to see the default `privileged`, `restricted` options and other PSACT options that I created; only admin users are able to do this. Standard users are the de facto standard for provisioning clusters, so we'll need to ensure that PSACT is properly work.

In other words, we need the ability to view global PSACTs and apply it to a cluster spec for standard users.

**To Reproduce**

1. Setup Rancher, create a standard user and login as that user.
2. Navigate to Cluster Management and note that PSACT is absent; you are unable to view global PSACTs.
3. Attempt to provision an RKE2 cluster and underneath the `Security` tab, click PSA.
     - Verify that you are unable to select any PSACT options.

**Result**
Standard users are unable to provision clusters with PSACT enabled due to being unable to view global PSACTs.

**Expected Result**

Standard users should be able to provision clusters with PSACT enabled and should be able to view global PSACTs.

**Comment by deniseschannon** [comment]
This needs to be broken into 2 separate issues. 

1. Enhancement - Creating a global role for Managing PSACTs
2. Bug - Ability to view global PSACTs and applying it to a cluster spec for standard users

**Comment by markusewalker** [comment]
Here is the enhancement issue: https://github.com/rancher/rancher/issues/40289

**Comment by snasovich** [comment]
Assigning @HarrisonWAffel to handle this.

I believe it should be about just adding the same access we have for PSPs to PSACTs in `role_data.go`, e.g.:
1. https://github.com/rancher/rancher/blob/1e993de98e9b303a4a8b1b57b583cc0f7ce16e4b/pkg/data/management/role_data.go#L50
2. https://github.com/rancher/rancher/blob/1e993de98e9b303a4a8b1b57b583cc0f7ce16e4b/pkg/data/management/role_data.go#L99
3. https://github.com/rancher/rancher/blob/1e993de98e9b303a4a8b1b57b583cc0f7ce16e4b/pkg/data/management/role_data.go#L116

Though addressing #2 may be part of https://github.com/rancher/rancher/issues/40289
Also, I'm not sure why we're adding it explicitly for `Create Cluster` role (#1) when it's already a part of `User` role (#3).

@cbron , we may need some help from your team on this. Please suggest/assign someone for this.

**Comment by crobby** [comment]
> 

I'm taking a look at this as well.  Those 3 lines in role_data appear to be right.  I'm checking to see if there are any other changes needed.

**Comment by snasovich** [comment]
@crobby , there are more hits on `podsecuritypolicytemplates` in `role_data.go` but they seem much more related to adding a separate role for handling permissions for actual CRD instances which is part of https://github.com/rancher/rancher/issues/40289.

**Comment by markusewalker** [comment]
Marking this as closed as standard users are now able to provision PSACT-enabled clusters. The internal test plan is keeping track of all the other open/blocking issues, but this issue can be closed.
